schema/claim.schema.json

```diff
@@ -66,6 +19,10 @@
       "description": "The date created, as an ISO-8601 Extended Format date string, as specified in the ECMAScript standard",
       "type": "string"
     },
+    "modified": {
```

Comment:
This is one area that I'm still grappling with... now that we separate the claim-result from the claim, with intentions that all updates/modifications are in the former, what scenarios will this claim object be modified (of, have a differing timestamp between `created` and `modified`)?

Reply:
Ah, as noted in the PR description: `Tools may choose to treat claim as immutable, but that isn't defined in the spec.`  This helps.  So, a claim may still be mutable and hence `modified` might then be utilized (different from `created`).

Reply:
I believe that this is how those fields are used:

* claim.created = The timestamp of when the claim for the entire bundle _installation_ was first created. This never ever changes, even when new revisions are created.
* claim.modified = The timestamp for when the bundle _installation_ was last upgraded, or invoked with a custom action that modifies resources (or uninstalled). This is what is bumped when we make new revisions.

If you think about what `porter list` displays today, we are able to use created and modified to tell when the bundle was installed and last executed using those two fields. That won't change.

What I meant by that comment was that there are two ways to think about the claim document:

1. The way that duffle treats it, where we constantly overwrite the document and lose the data from each revision
2. The way that we have been encouraged to treat the document going forward (to solve the outputs being lost problem), as snapshots of data (installation + revision) that should be stored separately.

My comment was referring to the second way of looking at claims, where the claim document after this change could treat the claim document as immutable because all of the data captured when an operation is executed doesn't change afterwards because everything unknown at the time of creation was moved into result. Further modifications to the bundle installation, like doing another operation such as upgrade, would result in a new document being created with a bumped revision of the claim.

I didn't put that comment about "immutable" in the spec because this mental model  (overwriting vs separate documents) isn't advocated for in the spec. It says you can do it either way (though we've found that overwriting really doesn't work well). I just mentioned it in the PR comment because we have been talking about the second model quite a bit with this change (moving result out of claim). Sorry if I caused extra confusion!

Reply:
Thank you for the detailed reasoning -- definitely helps me understand.